bin/test/i18n/resources/apps/external_sources/test_sync_up.rb

```diff
@@ -0,0 +1,73 @@
+require_relative '../../../../test_helper'
+require_relative '../../../../../i18n/resources/apps/external_sources/sync_up'
+
+describe I18n::Resources::Apps::ExternalSources::SyncUp do
+  let(:described_class) {I18n::Resources::Apps::ExternalSources::SyncUp}
+  let(:described_instance) {described_class.new}
+
+  around do |test|
+    FakeFS.with_fresh {test.call}
+  end
+
+  it 'inherits from I18n::Utils::SyncUpBase' do
+    assert_equal I18n::Utils::SyncUpBase, described_class.superclass
+  end
+
+  describe '#crowdin_project' do
+    let(:crowdin_project) {described_instance.send(:crowdin_project)}
+
+    let(:is_testing) {false}
+
+    before do
+      described_instance.stubs(:testing?).returns(is_testing)
+    end
+
+    it 'returns Crowdin project from config' do
+      assert_equal 'codeorg', crowdin_project
+    end
+
+    context 'when testing' do
+      let(:is_testing) {true}
+
+      it 'returns test Crowdin project from mapping' do
+        assert_equal 'codeorg-testing', crowdin_project
+      end
+    end
+  end
```

Comment:
This functionality is defined in `SyncUpBase`, right? Given that we're already verifying in a test that these classes inherit from that, I'm not sure we need to also verify this functionality explicitly. It's nice to keep tests focused on just the things about the class they're testing that make that class unique.

Reply:
You are right, they are more like integration tests to double-check that:
1. sync-up classes are configured correctly
2. i18n resources will be sync-upped to the right Crowdin projects.

I'll remove them then

Reply:
Yeah, and I do see there being value in having tests like that, I'd just prefer to see it implemented in the form of actual integration tests rather than trying to fit it in alongside the unit tests :)